biogeochem/EDPatchDynamicsMod.F90

```diff
@@ -1272,6 +1276,22 @@ subroutine set_patchno( currentSite )
        currentPatch => currentPatch%younger
     enddo
 
+    if(hlm_use_sp)then
```

Comment:
perhaps assign the non-sp type of patch number as an else condition so that it is not caculated unecessarily when not in sp mode.